cli/src/merge_tools/external.rs

```diff
@@ -310,13 +311,19 @@ fn run_mergetool_external_single_file(
             ExternalToolError::InvalidConflictMarkers { exit_status },
         ));
     }
-    // Update the file ids only, leaving the executable flags unchanged
-    let new_file_ids = if let Some(resolved) = new_file_ids.as_resolved() {
-        Merge::from_vec(vec![resolved.clone(); conflict.as_slice().len()])
-    } else {
-        new_file_ids
+
+    let new_tree_value = match new_file_ids.into_resolved() {
+        Ok(new_file_id) => Merge::normal(TreeValue::File {
+            id: new_file_id.unwrap(),
+            executable: conflict
+                .to_executable_merge()
+                .as_ref()
+                .and_then(Merge::resolve_trivial)
+                .copied()
+                .unwrap_or_default(),
```

Comment:
I think instead of using "not executable" as default here, we should make it an error and tell the user to run `jj file chmod` to choose whether it should be executable.

> Since exec bit has only two states, it can always be trivially resolved.

If a file was added as executable in one commit and added as non-executable in another commit and then you rebase on of them onto the other, it would result in a conflict in the executable bit. This is the kind of conflict I think we should inform the user of and make them resolve with `jj file chmod`. Another example is modify/delete conflicts where the modification changed the executable bit. Sorry that I wasn't clear about this before.

Does that make sense?

Reply:
> If a file was added as executable in one commit and added as non-executable in another commit and then you rebase on of them onto the other, it would result in a conflict in the executable bit. This is the kind of conflict I think we should inform the user of and make them resolve with `jj file chmod`. Another example is modify/delete conflicts where the modification changed the executable bit. Sorry that I wasn't clear about this before.

Do you mean we should change `.to_executable_merge()` to be tri-state? An absent file is interpreted as non-executable, so `[exec, absent, no-exec]` can be resolved to `exec`. I assumed the rationale here is that setting exec bit is an explicit operation, so `[(add file, set exec), (), (add file)]` can be merged as `(add file, set exec)`.

Reply:
> Do you mean we should change `.to_executable_merge()` to be tri-state?

It already is:

```
match value.to_executable_merge().map(|merge| merge.resolve_trivial()) {
  None => // some sides are trees or symlinks
  Some(Some(true)) => // executable
  Some(Some(false)) => // not executable
  Some(None) => // conflict in executable bit
}
```

It's for the last case I think we should tell the user to run `jj file chmod` first. Or, even better, have `jj resolve` ask the user whether they want the file to be executable.

Reply:
No, you're not misunderstanding. I was just blind. I really thought `to_executable_merge()` returned a `Option<Merge<Option<bool>>>` even after looking at it. I feel like it should return that type but I haven't checked what other consequences that would have.

Reply:
Ok, I'll take a look then. Perhaps, `jj resolve` will check if executable can be merged first, and error out immediately.

Reply:
Inserted patches to detect `[no-exec, exec, absent]` conflicts.